daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala

```diff
@@ -309,131 +302,103 @@ private[lf] object Anf {
           }
         // It's also safe to perform ANF for applications of a single argument.
         if (safeFunc || args.size == 1) {
-          transformMultiApp[A](depth, env, func, args.toArray, k)(transform)
+          transformMultiApp(depth, env, func, args.toArray, k)(transform)
         } else {
-          transformMultiAppSafely[A](depth, env, func, args.toArray, k)(transform)
+          transformMultiAppSafely(depth, env, func, args.toArray, k)(transform)
         }
 
-      case source.SEMakeClo(fvs0, arity, body0) =>
+      case source.SEMakeClo(fvs0, arity, body) =>
         val fvs = fvs0.map((loc) => makeRelativeL(depth)(makeAbsoluteL(env, loc)))
-        Bounce { () =>
-          val depth0 = DepthA(0)
-          val env0 = initEnv
-          flattenExp(depth0, env0, body0) { body =>
-            transform(depth, target.SEMakeClo(fvs.toArray, arity, body), k)
-          }
+        val depth0 = DepthA(0)
+        val env0 = initEnv
+        flattenExp(depth0, env0, body) { body =>
```

Comment:
`DepthA(0)` could be declared globally as `initEnv` to avoid necessary allocation.
Furthermore why not inlining this two definitions (I let you final judge) 

```suggestion
        val depth0 = DepthA(0)
        val env0 = initEnv
        flattenExp(depth0, env0, body) { body =>
```